pvlib/iotools/srml.py

```diff
@@ -0,0 +1,187 @@
+                      for flag in columns[1::2]}
+    data = data.rename(columns=flag_label_map)
+
+    # Mask data marked with quality flag 99 (bad or missing data)
+    for col in columns[::2]:
+        missing = data[col + '_flag'] == 99
+        data[col] = data[col].where(~(missing), np.NaN)
+    return data
+
+
+def map_columns(col):
+    """Map data element numbers to pvlib names.
+
+    Parameters
+    ----------
+    col: str
+        Column label to be mapped.
+
+    Returns
+    -------
+    str
+        The pvlib label if it was found in the mapping,
+        else the original label.
+
+    """
+    if col.startswith('7'):
+        # spectral data
+        try:
+            return variable_map[col]
+        except KeyError:
+            return col
+    try:
+        variable_name = variable_map[col[:3]]
+        variable_number = col[3:]
+        return variable_name + '_' + variable_number
+    except KeyError:
+        return col
+
+
+def format_index(df, year):
+    """ Create a datetime index from day of year, and time columns.
+
+    Parameters
+    ----------
+    df: pd.Dataframe
+        The srml data to reindex.
+    year: int
+        The year of the file
+
+    Returns
+    -------
+    df: pd.Dataframe
+        The Dataframe with a datetime index applied.
+
+    """
+    df_doy = df[df.columns[0]]
+    # Times are expressed as integers from 1-2400, we convert to 0-2359 by
+    # subracting one and then correcting the minutes at each former hour.
+    df_time = df[df.columns[1]] - 1
+    fifty_nines = df_time % 100 == 99
+    times = df_time.where(~fifty_nines, df_time - 40)
+
+    times = times.apply(lambda x: '{:04.0f}'.format(x))
+    doy = df_doy.apply(lambda x: '{:03.0f}'.format(x))
+    dts = pd.to_datetime(str(year) + '-' + doy + '-' + times,
+                         format='%Y-%j-%H%M')
+    df.index = dts
+    df = df.tz_localize('Etc/GMT+8')
+    return df
+
+
+def read_srml_month_from_solardat(station, year, month, filetype='PO'):
+    """Request a month of SRML[1] data from solardat and read it into
+    a Dataframe.
+
+    Parameters
+    ----------
+    station: str
+        The name of the SRML station to request.
+    year: int
+        Year to request data for
+    month: int
+        Month to request data for.
+    filetype: string
+        SRML file type to gather. 'RO' and 'PO' are the
+        only minute resolution files.
+    Returns
+    -------
+    data: pd.DataFrame
+        One month of data from SRML.
+
+    References
+    ----------
+    [1] University of Oregon Solar Radiation Measurement Laboratory
+        `http://solardat.uoregon.edu/ <http://solardat.uoregon.edu/>`_
+
+    """
+    file_name = "{station}{filetype}{year:02d}{month:02d}.txt".format(
+        station=station,
+        filetype=filetype,
+        year=year % 100,
+        month=month)
+    url = "http://solardat.uoregon.edu/download/Archive/"
+    data = read_srml(url + file_name)
+    return data
```

Comment:
empty line at end of file

Reply:
flake8 complains about this. with
`srml.py:185:1: W391 blank line at end of file`
Should I add it anyway? From what i can tell, pep8 just wants a newline at the end of the file and not an extra blank line.